server/src/item.rs

```diff
@@ -72,7 +84,22 @@ impl Item {
 }
 
 impl Item {
-    pub fn path(&self) -> ObjectPath<'_> {
-        self.path.as_ref()
+    pub fn new(
+        item: portal::Item,
+        locked: bool,
+        manager: Arc<Mutex<ServiceManager>>,
+        collection_path: OwnedObjectPath,
+        n_items: i32,
```

Comment:
this is the index, not n_items, also should be usize, not i32.

Reply:
Yup, done. Thanks!